neurons/validator.py

```diff
@@ -390,31 +390,35 @@ def update_models(self):
                             try:
                                 uid_last_retried_evaluation[uid] = dt.datetime.now()
 
-                                # Redownload this model and schedule it for eval even if it isn't updated by the sync.
+                                # Redownload this model and schedule it for eval even if it hasn't changed.
+                                # Still respect the eval block delay so that previously top uids can't bypass it.
                                 hotkey = metagraph.hotkeys[uid]
-                                asyncio.run(
-                                    self.model_updater.sync_model(hotkey, force=True)
+                                should_retry = self.model_updater.sync_model(
+                                    hotkey,
+                                    metagraph.block.item(),
```

Comment:
Note that here we use the deep copied metagraph we took for checking the top uids to keep alignment.

Below in the 'regular' case we use the self.metagraph which will be more up to date since it can get refreshed by the metagraph syncer out of this thread.

Reply:
Added a comment about this since if it is worth a comment in the PR probably worth a comment in the code.